### PR TITLE
SAMZA-1480: TaskStorageManager improperly initializes changelog consu…

### PR DESCRIPTION
**Pull request by jmakes**
…mer position when restoring a store from disk

### CODE REVIEW

**Comment by jmakes** [comment]
@prateekm early patch in case we need a hot fix.
Still needs unit tests. 
Will try to get to those Friday

File: samza-core/src/main/scala/org/apache/samza/storage/TaskStorageManager.scala [hunk 1]
```diff
@@ -228,11 +230,28 @@ class TaskStorageManager(
 
     for ((storeName, systemStream) <- changeLogSystemStreams) {
       val systemStreamPartition = new SystemStreamPartition(systemStream, partition)
+      val admin = systemAdmins.getOrElse(systemStream.getSystem,
+        throw new SamzaException("Unable to get systemAdmin for store " + storeName + " and systemStream" + systemStream))
```
Comment (prateekm) [inline samza-core/src/main/scala/org/apache/samza/storage/TaskStorageManager.scala]: Minor: 
"system admin for store: " 
"and system stream: "

File: samza-core/src/main/scala/org/apache/samza/storage/TaskStorageManager.scala [hunk 2]
```diff
@@ -228,11 +230,28 @@ class TaskStorageManager(
...
-        Option(fileOffset.get(systemStreamPartition))
-          .getOrElse(changeLogOldestOffsets
-            .getOrElse(systemStream, throw new SamzaException("Missing a change log offset for %s." format systemStreamPartition)))
+      val fileOffset = fileOffsets.get(systemStreamPartition)
+      val oldestOffset = changeLogOldestOffsets
+        .getOrElse(systemStream, throw new SamzaException("Missing a change log offset for %s." format systemStreamPartition));
+      val offset = if (fileOffset != null) {
+        // File offset was the last message written to the changelog that is also reflected in the store,
+        // so we start with the NEXT offset
+        val resumeOffset = admin.getOffsetsAfter(Map(systemStreamPartition -> fileOffset).asJava).get(systemStreamPartition)
```
Comment (prateekm) [inline samza-core/src/main/scala/org/apache/samza/storage/TaskStorageManager.scala]: Minor: IMHO it's worth extracting this as a private helper method (e.g. getStartingOffset(oldestOffset, fileOffset)) with documentation + tests.

File: samza-core/src/main/scala/org/apache/samza/storage/TaskStorageManager.scala [hunk 3]
```diff
@@ -228,11 +230,28 @@ class TaskStorageManager(
...
+        // If the offset we plan to use is older than the oldest offset, just use the oldest offset.
```
Comment (prateekm) [inline samza-core/src/main/scala/org/apache/samza/storage/TaskStorageManager.scala]: Minor: Prefer moving this comment to the else block.

File: samza-core/src/main/scala/org/apache/samza/storage/TaskStorageManager.scala (hunk 2 above)
Comment (jmakes) [reply samza-core/src/main/scala/org/apache/samza/storage/TaskStorageManager.scala]: Totally agree. I've actually already done this on my local copy. Will update the PR when the unit tests are done.

**Pull request closed without merging**